storage-node-v2/src/command-base/ApiCommandBase.ts

```diff
@@ -0,0 +1,135 @@
+import { Command, flags } from '@oclif/command'
+import { createApi } from '../services/runtime/api'
+import { getAccountFromJsonFile, getAlicePair } from '../services/runtime/accounts'
+import { KeyringPair } from '@polkadot/keyring/types'
+import { ApiPromise } from '@polkadot/api'
+import logger from '../services/logger'
+import ExitCodes from './ExitCodes'
+import { CLIError } from '@oclif/errors'
+import { Input } from '@oclif/parser'
+
+/**
+ * Parent class for all runtime-based commands. Defines common functions.
+ */
+export default abstract class ApiCommandBase extends Command {
+  private api: ApiPromise | null = null
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    dev: flags.boolean({ char: 'm', description: 'Use development mode' }),
+    apiUrl: flags.string({
+      char: 'u',
+      description: 'Runtime API URL. Mandatory in non-dev environment. Default is ws://localhost:9944',
+    }),
+    keyfile: flags.string({
+      char: 'k',
+      description: 'Key file for the account. Mandatory in non-dev environment.',
```

Comment:
```suggestion
      description: 'Key file for the account. Mandatory in non-dev environment.',
      required: (flags) => {   "true if dev flag is not set"  }
```
I'm not familiar with oclif but can the value for `required` be a function that evaluates to a bool depending on what other flags were set? With `meow` we do something like this: https://github.com/Joystream/joystream/blob/8b1f4a5df3b1cdb903730e1d51c60e6a3d033cef/storage-node/packages/colossus/bin/cli.js#L59

Reply:
Not sure it's possible.